src/Core/Layer/Layer.js

```diff
@@ -1,158 +1,151 @@
-        if (obj.parent && obj.material) {
-            return {
-                element: obj,
-                parent: obj.parent,
+import * as THREE from 'three';
+import { STRATEGY_MIN_NETWORK_TRAFFIC } from '../../Layer/LayerUpdateStrategy';
+
+class Layer extends THREE.EventDispatcher {
+    /**
+     * Don't use directly constructor to instance a new Layer. Instead, use
+     * another available type of Layer, implement a new one inheriting from this
+     * one or use {@link View#addLayer}.
+     *
+     * @constructor
+     * @protected
+     *
+     * @param {string} id - The id of the layer, that should be unique. It is
+     * not mandatory, but an error will be emitted if this layer is added a
+     * {@link View} that already has a layer going by that id.
+     * @param {string} type - The type of the layer, used to determine
+     * operations to do on a layer later in the rendering loop. There are three
+     * type of layers in itowns:
+     * <ul>
+     *  <li><code>color</code>, used for simple layer containing a texture</li>
+     *  <li><code>elevation</code>, used for adding an elevation to the globe or
+     *  plane the layer is attached to</li>
+     *  <li><code>geometry</code>, used for complex layer containing meshes,
+     *  like a WFS layer with extruded buildings</li>
+     * <ul>
+     * @param {Object} [config] - Optional configuration, all elements in it
+     * will be merged as is in the layer. For example, if the configuration
+     * contains three elements <code>name, protocol, extent</code>, these
+     * elements will be available using <code>layer.name</code> or something
+     * else depending on the property name.
+     *
+     * @example
+     * // Add and create a new Layer
+     * const newLayer = view.addLayer({options});
+     *
+     * // Change layer's visibilty
+     * const layerToChange = view.getLayers(layer => layer.id == 'idLayerToChange')[0];
+     * layerToChange.visible = false;
+     * view.notifyChange(); // update viewer
+     *
+     * // Change layer's opacity
+     * const layerToChange = view.getLayers(layer => layer.id == 'idLayerToChange')[0];
+     * layerToChange.opacity = 0.5;
+     * view.notifyChange(); // update viewer
+     *
+     * // Listen properties
+     * const layerToListen = view.getLayers(layer => layer.id == 'idLayerToListen')[0];
+     * layerToListen.addEventListener('visible-property-changed', (event) => console.log(event));
+     * layerToListen.addEventListener('opacity-property-changed', (event) => console.log(event));
+     */
+    constructor(id, type, config = {}) {
+        super();
+
+        Object.assign(this, config);
+
+        Object.defineProperty(this, 'id', {
+            value: id,
+            writable: false,
+        });
+
+        Object.defineProperty(this, 'type', {
+            value: type,
+            writable: false,
+        });
+
+        // Default properties
+        this.options = config.options || {};
```

Comment:
It would be necessary, to write `_` before private property name.
Could you tell me the properties that make sense to be accessible? 

by example` this.isAttached` is not supposed to be modifiable by the user

Reply:
So this line comes from `View`, so I kinda let it as it is. For private property name, I'm okay to write it with `_`, but I don't know a way to keep a property modifiable by us but not by the user.